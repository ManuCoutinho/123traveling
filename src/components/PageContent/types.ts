export type PageContentProps = {
	city?: string
	country?: string
	language?: string
	text?: string
}
