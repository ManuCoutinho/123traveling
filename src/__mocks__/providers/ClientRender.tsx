/* eslint-disable react/display-name */

import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import { ChakraProvider } from '@chakra-ui/react'

const createTestQueryClient = () =>
	new QueryClient({
		defaultOptions: {
			queries: {
				retry: false
			}
		}
	})

export function renderWithClient(children: React.ReactNode) {
	const testQueryClient = createTestQueryClient()
	const { rerender, ...result } = render(
		<ChakraProvider>
			<QueryClientProvider client={testQueryClient}>
				{children}
			</QueryClientProvider>
		</ChakraProvider>
	)
	return {
		...result,
		rerender: (rerenderUi: React.ReactNode) =>
			rerender(
				<QueryClientProvider client={testQueryClient}>
					{rerenderUi}
				</QueryClientProvider>
			)
	}
}

export const customWrapper = () => {
	const testQueryClient = createTestQueryClient()

	return ({ children }: { children: React.ReactNode }) => (
		<QueryClientProvider client={testQueryClient}>
			{children}
		</QueryClientProvider>
	)
}
