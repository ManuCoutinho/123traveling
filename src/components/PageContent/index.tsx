import { Box, Grid, Text, useColorMode } from '@chakra-ui/react'
import { BoxInfo } from '../BoxInfo'
import { PageContentProps } from './types'

const PageContent: React.FC<PageContentProps> = ({
	city,
	country,
	language,
	text
}) => {
	const { colorMode } = useColorMode()
	const textColor = colorMode === 'light' ? 'gray.600' : 'whiteAlpha.700'
	return (
		<Grid
			templateColumns='repeat(auto-fit, minmax(250px, 1fr))'
			gap={[4, 6, 8, 12]}
			my={16}
			px={[2, 4, 6, 14]}
			mx='auto'>
			<Box>
				<Text
					fontSize={['sm', 'md', 'lg', 'xl']}
					lineHeight='tall'
					color={textColor}>
					{text}
				</Text>
			</Box>
			<BoxInfo country={country} language={language} city={city} />
		</Grid>
	)
}
export default PageContent
