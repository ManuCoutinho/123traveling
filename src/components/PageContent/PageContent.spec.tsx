import { screen } from '@testing-library/react'
import { renderWithClient } from '__mocks__/providers/ClientRender'
import PageContent from '.'

const dataMock = {
	city: 'cairo',
	country: 'egito',
	language: '5',
	text: 'text placeholder'
}
describe('<PageContent/>', () => {
	it('should render PageContent correctly ', () => {
		renderWithClient(<PageContent {...dataMock} />)
		expect(screen.getByText(dataMock.city)).toBeInTheDocument()
		expect(screen.getByText(dataMock.country)).toBeInTheDocument()
		expect(screen.getByText(dataMock.language)).toBeInTheDocument()
		expect(screen.getByText(dataMock.text)).toBeInTheDocument()
	})
	it('should match to snapshot', () => {
		const { container } = renderWithClient(<PageContent {...dataMock} />)
		expect(container).toMatchSnapshot()
	})
})
