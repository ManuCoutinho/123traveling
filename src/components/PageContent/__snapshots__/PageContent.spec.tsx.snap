// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContent/> should match to snapshot 1`] = `
<div>
  <div
    class="css-j506mb"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-c4euvy"
      >
        text placeholder
      </p>
    </div>
    <div
      class="chakra-stack css-ez4v9i"
    >
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-k2gyu4"
        >
          egito
        </p>
        <div
          class="chakra-stack css-84zodg"
        >
          <p
            class="chakra-text css-z9fjiz"
          >
            pa√≠ses
          </p>
          <span />
        </div>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-k2gyu4"
        >
          5
        </p>
        <div
          class="chakra-stack css-84zodg"
        >
          <p
            class="chakra-text css-z9fjiz"
          >
            idiomas
          </p>
          <span>
            <span>
              <svg
                aria-hidden="true"
                color="#47585B"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-width="2"
                style="color: rgb(71, 88, 91);"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-k2gyu4"
        >
          cairo
        </p>
        <div
          class="chakra-stack css-84zodg"
        >
          <p
            class="chakra-text css-z9fjiz"
          >
            cidades +100
          </p>
          <span>
            <span>
              <svg
                aria-hidden="true"
                color="#47585B"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-width="2"
                style="color: rgb(71, 88, 91);"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
